### PR TITLE
new challenge - function to determine whether a given number is prime

### PR DESCRIPTION
**Pull request by RandomProjects-db**
This PR introduces the following changes:

is_prime Function:
Determines whether a given number is prime.
Handles edge cases like 0, 1, and negative numbers.
Raises an AssertionError for invalid inputs (non-integer types).
Custom Assertion in Tests:
Added assert_prime(expected, number) for consistent and informative test assertions.
Unit Tests:
One assertion per test case.
Comprehensive test coverage for valid inputs, edge cases, and invalid inputs.

### CODE REVIEW

**Review by ziadahanass — CHANGES_REQUESTED** [review]

File: solutions/tests/test_is_prime.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    def test_edge_case_one(self):
+        """Test case for one."""
+        assert_prime(False, 1)
+
+    def test_invalid_string(self):
+        """Test case for an invalid string input."""
+        with self.assertRaises(AssertionError):
+            is_prime("eleven")
+
```
[the remark this answers is not in the document]
Comment (RandomProjects-db) [reply solutions/tests/test_is_prime.py:75]: fixed!

File: solutions/is_prime.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+"""
```
[the remark this answers is not in the document]
Comment (RandomProjects-db) [reply solutions/is_prime.py:1]: fixed.

File: solutions/is_prime.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+is_prime.py
+
+This module provides a function to determine whether a given number is prime.
+A prime number is a natural number greater than 1 that is not divisible by any
+number other than 1 and itself.
```
[the remark this answers is not in the document]
Comment (RandomProjects-db) [reply solutions/is_prime.py:6]: fixed.

File: solutions/is_prime.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+Functions:
+- is_prime(number):
+  Determines whether the input number is a prime number.
+"""
+
+
+def is_prime(number):
+    """
+    Determines if a given number is prime.
```
[the remark this answers is not in the document]
Comment (RandomProjects-db) [reply solutions/is_prime.py:20]: fixed

File: solutions/is_prime.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (RandomProjects-db) [reply solutions/is_prime.py:6]: for more clarity

File: solutions/tests/test_is_prime.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ziadahanass) [reply solutions/tests/test_is_prime.py:78]: I am gonna merge it

**Review by RandomProjects-db — COMMENTED** [review]
fixed

**Review by ziadahanass — APPROVED** [review]
